### PR TITLE
Make Status read only

### PR DESCRIPTION
**Pull request by Jell**
So we have this use case where our developers have read-only access to our databases, and want to be able to run `dbmate status` before running a migration as part of our CI/CD pipeline.

Surprisingly, `dbmate status` has the potential side-effect of creating the migrations table if it's not there, which fails for a read only user.

This change makes the Status command read only by checking if the table exists, and returning all migrations as pending if it doesn't (and works as expected otherwise).

### CODE REVIEW

**Comment by Jell** [comment]
anything I could do to help move this forward?

**Comment by Enrico204** [comment]
Interesting. I agree, status should be R/O, there is no need for it to create a table

File: pkg/driver/sqlite/sqlite_test.go [hunk 1]
```diff
@@ -233,6 +233,7 @@ func TestSQLiteSelectMigrations(t *testing.T) {
 	// test limit param
 	migrations, err = drv.SelectMigrations(db, 1)
 	require.NoError(t, err)
+
```
Comment (Enrico204) [inline pkg/driver/sqlite/sqlite_test.go:236]: You may want to remove this space, to avoid creating conflicts where there is no need to :-)

Comment (Jell) [reply pkg/driver/sqlite/sqlite_test.go]: fixed!

**Comment by amacneil** [comment]
Would be good to add a test making sure the table doesn’t get created before we merge

**Comment by Jell** [comment]
> Would be good to add a test making sure the table doesn’t get created before we merge

Done! https://github.com/amacneil/dbmate/pull/242/commits/7c76b9b3b15ef1ea30c0dd3c90ffb6b9745003ca

**Comment by amacneil** [comment]
Thanks!

**Pull request merged**